addons/controllers/antrea/antreaconfig_controller.go

```diff
@@ -64,31 +65,31 @@ func (r *AntreaConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
+	annotations := antreaConfig.GetAnnotations()
+	if _, ok := annotations[constants.TKGAnnotationTemplateConfig]; ok {
+		log.Info(fmt.Sprintf("resource '%v' is a config template. Skipping reconciling", req.NamespacedName))
+		return ctrl.Result{}, nil // TODO (adduarte) this should return err not nil to use backoff algorithms.
+	}
+
 	// deep copy AntreaConfig to avoid issues if in the future other controllers where interacting with the same copy
 	antreaConfig = antreaConfig.DeepCopy()
 
-	// get the parent cluster name from owner reference
-	// if the owner reference doesn't exist, use the same name as config CRD
-	clusterNamespacedName := req.NamespacedName
-	cluster := &clusterapiv1beta1.Cluster{}
-	for _, owner := range antreaConfig.OwnerReferences {
-		if owner.Kind == constants.ClusterKind {
-			clusterNamespacedName.Name = owner.Name
-			break
-		}
-	}
+	cluster, err := cutil.GetOwnerCluster(ctx, r.Client, antreaConfig, req.Namespace, constants.AntreaDefaultRefName)
 
-	// verify that the cluster related to config is present
-	if err := r.Client.Get(ctx, clusterNamespacedName, cluster); err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Info(fmt.Sprintf("Cluster '%s'not found in '%s'", clusterNamespacedName.Name, clusterNamespacedName.Namespace))
-			return ctrl.Result{}, nil
+	if err != nil {
+		if apierrors.IsNotFound(err) && cluster != nil {
+			log.Info(fmt.Sprintf("'%s/%s' is listed as owner reference but could not be found",
+				cluster.Namespace, cluster.Name))
+			return ctrl.Result{}, err
 		}
-
-		log.Error(err, fmt.Sprintf("unable to fetch cluster '%s' in '%s'", clusterNamespacedName.Name, clusterNamespacedName.Namespace))
+		log.Error(err, "could not determine owner cluster")
 		return ctrl.Result{}, err
 	}
 
+	if cluster == nil {
+		return ctrl.Result{}, err // retry until corresponding cluster is found
```

Comment:
If cluster is nil will err be nil ? If err is nil anyways it will return from above if err != nil.

Reply:
There is a logical chance that we have cluster=nil and err=nil, can't think of one but it would happen if the function ```err := k8sClient.Get(ctx, client.ObjectKeyFromObject(cluster), cluster)`` returns cluster=nil and err = nil.  
Basically GetOwnerCluster has no way of determining what it means cluster=nil and err=nil so it just kicks it up (to here) and that situation is probably an error. Maybe we should handle it differently then (since err=nil).

Reply:
GetOwnerCluster will never return cluster==nil and err == nil, thus this is dead code, will remove